tests/lib/rules/no-else-return.js

```diff
@@ -30,22 +30,79 @@ ruleTester.run("no-else-return", rule, {
+            output: "function foo() { if (true) { if (false) { if (true) return x;  w = y;  } else { w = x; } } else { return z; } }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { if (false) { if (true) return x; else return y; } } else { return z; } }",
+            output: "function foo() { if (true) { if (false) { if (true) return x; return y; } } else { return z; } }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "ReturnStatement" }]
+        },
+        {
+            code: "function foo() { if (true) { if (false) { if (true) return x; else return y; } return w; } else { return z; } }",
+            output: "function foo() { if (true) { if (false) { if (true) return x; return y; } return w; }  return z;  }",
+            errors: [
+                { message: "Unnecessary 'else' after 'return'.", type: "ReturnStatement" },
+                { message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }
+            ]
+        },
+        {
+            code: "function foo() { if (true) { if (false) { if (true) return x; else return y; } else { w = x; } } else { return z; } }",
+            output: "function foo() { if (true) { if (false) { if (true) return x; return y; }  w = x;  } else { return z; } }",
+            errors: [
+                { message: "Unnecessary 'else' after 'return'.", type: "ReturnStatement" },
+                { message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }
+            ]
+        },
         {
             code: "function foo() {if (x) { return true; } else if (y) { return true; } else { notAReturn(); } }",
+            output: "function foo() {if (x) { return true; } else if (y) { return true; }  notAReturn();  }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (foo) return bar; else (foo).bar(); }",
+            output: "function foo() { if (foo) return bar; (foo).bar(); }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "ExpressionStatement" }]
+        },
+        {
+            code: "function foo() { if (foo) return bar \nelse { [1, 2, 3].map(foo) } }",
+            output: "function foo() { if (foo) return bar \nelse { [1, 2, 3].map(foo) } }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }]
+        },
+        {
+            code: "function foo() { if (foo) return bar \nelse { baz() } \n[1, 2, 3].map(foo) }",
+            output: "function foo() { if (foo) return bar \nelse { baz() } \n[1, 2, 3].map(foo) }",
+            errors: [{ message: "Unnecessary 'else' after 'return'.", type: "BlockStatement" }]
```

Comment:
Can you add test cases like the two above, but with a semicolon?

```js
"function foo() { if (foo) return bar; \nelse { [1, 2, 3].map(foo) } }"
"function foo() { if (foo) return bar \nelse { baz(); } \n[1, 2, 3].map(foo) }"
```

Both of these cases should result in a fix being applied.